docs/deployment/kubernetes/eks.md

```diff
@@ -0,0 +1,191 @@
+---
+title: EKS + Helm
+sidebar_position: 3
+---
+
+This guide will help you set up a Quickwit cluster on EKS with the correct S3 permissions.
+
+## Prerequisites
+- Running Elastic Kubernetes cluster (EKS)
+- `kubectl`
+- Permission to create the IAM role and Policies
+- AWS CLI
+- `eksctl` if you don't have an IAM OIDC provider for your cluster.
+
+## Set up
+
+Let's use the following environment variables:
+
+```bash
+export NAMESPACE=qw-tutorial
+export EKS_CLUSTER=qw-cluster
+export S3_BUCKET={your-bucket}
+export SERVICE_ACCOUNT_NAME=qw-sa
+export REGION={your-region}
+export CLUSTER_ID={your-cluster-id}
+```
+
+Create the namespace for our playground:
+
+```bash
+kubectl create ns ${NAMESPACE}
+```
+
+And set this namespace as the default one:
+
+```bash
+kubectl config set-context --current --namespace=${NAMESPACE}
+```
+
+
+### Create IAM OIDC provider if you don't have one
+
+To check if you have one provider for your EKS cluster, just run:
+
+```bash
+aws iam list-open-id-connect-providers
+```
+
+If you have one, you will get a response similar to this one:
+
+```json
+{
+    "OpenIDConnectProviderList": [
+        {
+            "Arn": "arn:aws:iam::(some-ID):oidc-provider/oidc.eks.{your-region}.amazonaws.com/id/{your-cluster-id}"
+        }
+    ]
+}
+```
+
+If you don't, run the following command:
+
+```bash
+eksctl utils associate-iam-oidc-provider --cluster ${EKS_CLUSTER} --approve
+```
+
+You can run again `aws iam list-open-id-connect-providers` to get the ARN of the provider.
+
+### Create an IAM policy
+
+You need to set the following policy to allow Quickwit to access your S3 bucket.
+
+Then create the policy using the AWS CLI:
+
+```bash
+cat > s3-policy.json <<EOF
+{
+    "Statement": [
+        {
+            "Action": [
+                "s3:AbortMultipartUpload",
+                "s3:DeleteObject",
+                "s3:GetObject",
+                "s3:ListMultipartUploadParts",
+                "s3:PutObject"
```

Comment:
```
If the object that you request doesn’t exist, the error that Amazon S3 returns depends on whether you also have the s3:ListBucket permission.
   - If you have the s3:ListBucket permission on the bucket, Amazon S3 returns an HTTP status code 404 Not Found error.
   - If you don’t have the s3:ListBucket permission, Amazon S3 returns an HTTP status code 403 Access Denied error.
```
(extract from https://docs.aws.amazon.com/AmazonS3/latest/API/API_GetObject.html)
we probably need `s3:ListBucket` (or to handle errors differently), at least when initializing the metastore for the first time